docs/sections/contributor_guide/developer_setup.rst

```diff
@@ -1,5 +1,147 @@
+
+   .. code::
+
+     cat <<EOF >~/.condevrc
+     source ~/.bashrc
+     EOF
+
+
+Using a ``bash`` development shell
+==================================
+
+In an active development shell, the following ``make`` targets are available and act on all ``.py`` files under ``src/``:
+
+Formats Python code with black, imports with isort, docstrings with docformatter, and .jsonschema documents with jq.
+
++---------------------+------------------------------------------------------------+
+| Command             |  Description                                               |
++=====================+============================================================+
+| ``make format``     | Formats python code  with `black`_,                        |
+|                     | imports with `isort`_,                                     |
+|                     |                                                            |  
+|                     | docstrings with `docformatter`_,                           |
+|                     | and ``.jsonschema`` documents with `jq`_                   |
++---------------------+------------------------------------------------------------+
+| ``make lint``       | Lint with `pylint`_                                        |
+|                     |                                                            |
++---------------------+------------------------------------------------------------+
+| ``make type check`` | Typecheck with `mypy`_                                     |
++---------------------+------------------------------------------------------------+
+| ``make unittest``   | Run unit tests and report coverage with `pytest`_ and      |
+|                     | `coverage`_                                                |
++---------------------+------------------------------------------------------------+
+| ``make test``       | Equivalent to                                              |
+|                     | ``make lint && make typecheck && make unittest``           |
+|                     |                                                            |
+|                     | Checks defined CLI scripts                                 |
++---------------------+------------------------------------------------------------+
+
+
+Note that ``make format`` is never run automatically, to avoid reformatting under-development code in a way that might surprise the developer. A useful development idiom is to periodically run
+``make format && make test`` to perform a full code-quality sweep through the code.
+
+The ``make test`` command is also automatically executed when ``conda`` builds a ``uwtools`` package, so it is important to periodically run these tests during development and, crucially, before merging changes,
+to ensure that the tests will pass when CI builds the ``workflow-tools`` code.
```

Comment:
nitpick: There are still some manually broken lines here that should be combined, e.g. 87 + 88 and 90 + 91. Thanks for what you've already done, though!